src/arch/ArchHooks/ArchHooks_Win32Static.cpp

```diff
@@ -22,30 +22,35 @@ static bool g_bTimerInitialized;
  * QueryPerformanceCounter and QueryPerformanceFrequency expect
  * a LARGE_INTEGER, which is a union. These functions store data
  * in the QuadPart of the LARGE_INTEGER, which is a 64-bit integer. */
-LARGE_INTEGER g_liFrequency, g_liStartTime, g_liCurrentTime;
+namespace {
+    LARGE_INTEGER g_liFrequency;
+    LARGE_INTEGER g_liCurrentTime;
+}
 
 static void InitTimer()
 {
-	if( g_bTimerInitialized )
+	if( g_bTimerInitialized ) {
 		return;
+	}
+	
 	g_bTimerInitialized = true;
 
-	// Grab important information for QPC during the timer initialization
-	QueryPerformanceCounter(&g_liStartTime);
+	// Retrieve the number of ticks per second.
 	QueryPerformanceFrequency(&g_liFrequency);
 }
 
 std::int64_t ArchHooks::GetMicrosecondsSinceStart(bool bAccurate)
 {
-	// Make sure the timer is initialized
-	if (!g_bTimerInitialized)
+	// Make sure the timer is initialized.
+	if (!g_bTimerInitialized) {
 		InitTimer();
+	}
 
-	// Get the current time
+	// Get the current time.
 	QueryPerformanceCounter(&g_liCurrentTime);
 
 	// Calculate the elapsed time in microseconds.
-	return ((g_liCurrentTime.QuadPart - g_liStartTime.QuadPart) * 1000000.0) / g_liFrequency.QuadPart;
+	return (g_liCurrentTime.QuadPart * 1000000) / g_liFrequency.QuadPart;
```

Comment:
Do we lose any precision without the implicit float conversion?

Reply:
We do not lose any precision without the float conversion.

QueryPerformanceCounter is the current tick count; QueryPerformanceCounter is the number of ticks per second. In almost all cases of a x86_64 CPU that is 10mhz.

The time value in ticks is being multiplied by 1000000 so we return a value of microseconds. If we don't multiply by 1000000 we get an answer in seconds.

The above value is then divided by the number of ticks per second.

Odd values for QueryPerformanceFrequency aren't possible.

Reply:
For further confirmation that this is safe without any precision loss, we can evaluate the math happening.

10MHz is nearly ubiquitous for any x86 cpu with a high resolution timer, this gives a QueryPerformanceFrequency result of 10000000

thus 99-100% of users on a x86_64 cpu have this happening:
```
return (g_liCurrentTime.QuadPart * 1000000) / 10000000;
```

For a system with a 10mhz timer, let's assume the system is running for 3 days. QueryPerformanceCounter gives something like 2592000000000

2592000000000 * 1000000 = 2.592e+18

2.592e+18 / 10000000 = 259200000000

so in the end the vast majority of Windows users will effectively be doing 
```
return g_liCurrentTime.QuadPart / 10;
```